src/WorkflowManager/Common/Services/PayloadService.cs

```diff
@@ -109,14 +114,54 @@ public async Task<Payload> GetByIdAsync(string payloadId)
             return await _payloadRepository.GetByIdAsync(payloadId);
         }
 
-        public async Task<IList<Payload>> GetAllAsync(int? skip = null,
+        public async Task<IList<PayloadDto>> GetAllAsync(int? skip = null,
                                                       int? limit = null,
                                                       string? patientId = "",
                                                       string? patientName = "")
-            => await _payloadRepository.GetAllAsync(skip, limit, patientId, patientName);
+            => await CreatePayloadsDto(
+                await _payloadRepository.GetAllAsync(skip, limit, patientId, patientName)
+            );
+
+        public async Task<IList<PayloadDto>> GetAllAsync(int? skip = null, int? limit = null)
+            => await CreatePayloadsDto(await _payloadRepository.GetAllAsync(skip, limit));
+
+        private async Task<IList<PayloadDto>> CreatePayloadsDto(IList<Payload> payloads)
+        {
+            var dtos = new List<PayloadDto>();
+            if (payloads is null || payloads.Count == 0)
+            {
+                return dtos;
+            }
+
+            const string Completed = "Completed";
+            const string InProgress = "In Progress";
```

Comment:
Would it be better to have a separate constants file for these? Then they can be in their own static class named `PayloadStatusConstants`? In case other methods need it in the future. Just a suggestion anyway

Reply:
Yeah I'll make this change